src/esphome-main.ts

```diff
@@ -32,15 +39,15 @@ class ESPHomeMainView extends LitElement {
     }
     return html`
       <main>
-        <esphome-devices-list></esphome-devices-list>
+        <esphome-devices-list .viewMode=${this._isWide ? this.viewMode : ViewMode.Module}></esphome-devices-list>
       </main>
       <div class="esphome-header">
         <img
           src="https://esphome.io/_static/logo-text.svg"
           alt="ESPHome Logo"
         />
         <div class="flex"></div>
-        <esphome-header-menu .logoutUrl=${this.logoutUrl}></esphome-header-menu>
+        <esphome-header-menu .logoutUrl=${this.logoutUrl} .toggleViewMode=${this._toggleViewMode.bind(this)} .viewMode=${this.viewMode}></esphome-header-menu>
```

Comment:
maybe instead of passing a function consider dispatching a custom event and listening to that event.
Ref: https://github.com/esphome/dashboard/blob/2a59a7875908a5bda9e19b033e947a074b85e00e/src/components/remote-process.ts#L47

Reply:
Thanks for your feedback :tada:  I updated the code.